ui/components/multichain/token-list-item/price/percentage-and-amount-change/percentage-and-amount-change.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { BigNumber } from 'bignumber.js';
+import { isHexString, zeroAddress } from 'ethereumjs-util';
+import { Text, Box } from '../../../../component-library';
+import {
+  Display,
+  FontWeight,
+  TextColor,
+  TextVariant,
+} from '../../../../../helpers/constants/design-system';
+import {
+  getCurrentCurrency,
+  getSelectedAccountCachedBalance,
+  getTokensMarketData,
+} from '../../../../../selectors';
+import { getIntlLocale } from '../../../../../ducks/locale/locale';
+import { EtherDenomination } from '../../../../../../shared/constants/common';
+import { Numeric } from '../../../../../../shared/modules/Numeric';
+import {
+  getConversionRate,
+  getNativeCurrency,
+} from '../../../../../ducks/metamask/metamask';
+
+const renderPercentageWithNumber = (
+  value: string,
+  formattedValuePrice: string,
+  color: TextColor,
+) => {
+  return (
+    <Box display={Display.Flex}>
+      <Text
+        fontWeight={FontWeight.Normal}
+        variant={TextVariant.bodyMd}
+        color={color}
+        data-testid="token-increase-decrease-value"
+        style={{ whiteSpace: 'pre' }}
+        ellipsis
+      >
+        {formattedValuePrice}
+      </Text>
+      <Text
+        fontWeight={FontWeight.Normal}
+        variant={TextVariant.bodyMd}
+        color={color}
+        data-testid="token-increase-decrease-percentage"
+        ellipsis
+      >
+        {value}
+      </Text>
+    </Box>
+  );
+};
+
+export const PercentageAndAmountChange = ({
+  value,
+}: {
+  value: number | null | undefined;
+}) => {
+  const fiatCurrency = useSelector(getCurrentCurrency);
+  const locale = useSelector(getIntlLocale);
+  const balanceValue = useSelector(getSelectedAccountCachedBalance);
+  const conversionRate = useSelector(getConversionRate);
+  const nativeCurrency = useSelector(getNativeCurrency);
+  const marketData = useSelector(getTokensMarketData);
+
+  const balanceChange = useMemo(() => {
+    const percentage1d = marketData?.[zeroAddress()]?.pricePercentChange1d;
+
+    if (isHexString(balanceValue)) {
```

Comment:
Can we add some comments for the math usage here so maintainers feel comfortable knowing what's happening?

Reply:
done , just added some comments